peps/pep-0738.rst

```diff
@@ -0,0 +1,351 @@
+This is the same pattern used by CPython on Windows.
+
+Unlike iOS, Android allows dynamic libraries to be loaded from any location, so
+a directory tree containing co-located .py, .pyc and .so files can be handled by
+Python's standard importer.
+
+
+Standard library
+----------------
+
+A number of standard library modules will not be supported on Android because
+the underlying C APIs are not provided by the operating system:
+
+* ``crypt`` – Scheduled for removal in Python 3.13
+* ``curses`` and ``readline``
+* ``dbm.gnu`` and ``dbm.ndbm``
+* ``grp``, ``nis`` and ``spwd``
+
+Android does not support any part of the `System V IPC API
+<https://man7.org/linux/man-pages/man7/sysvipc.7.html>`__, so the
+``multiprocessing`` module will not work in its current form. However,
+subprocesses in general are allowed, with some limitations – see `App
+lifecycle`_.
+
+``tkinter`` and ``turtle`` will not be included, because that would require an
+Android build of Tk itself, which is not officially supported.
+
+
+Platform identification
+-----------------------
+
+``sys``
+'''''''
+
+``sys.platform`` will return ``"android"``. Although Android is based on Linux,
+it differs in enough significant ways that a separate name is justified.
+
+``platform``
+''''''''''''
+
+Most of the values returned by the ``platform`` module will match those returned
+by ``os.uname()``, with the exception of:
+
+* ``platform.system()`` - ``"Android"``, instead of the default ``"Linux"``
+
+* ``platform.release()`` - Android version number, as a string (e.g. ``"14"``),
+  instead of the Linux kernel version
+
+In addition, a ``platform.android_ver()`` method will be added, which returns a
+namedtuple containing the following:
+
+* ``release`` - Android version, as a string (e.g. ``"14"``)
+* ``api_level`` - Android API level, as an integer (e.g. ``34``)
+* ``min_api_level`` - Minimum API level this build of Python can run on, as
+  an integer (e.g. ``23``). This is the same as ``sys.getandroidapilevel``.
+* ``model`` - the model name of the device, as a string (e.g. ``"Pixel 7"``).
+
+``os``
+''''''
+
+``os.uname()`` will return the raw result of a POSIX ``uname()`` call. This will
+result in the following values:
+
+* ``sysname`` - ``"Linux"``
+
+* ``release`` - The Linux kernel version (e.g.
+  ``"5.10.157-android13-4-00003-gdfb1120f912b-ab10994928"``)
+
+This approach treats the ``os`` module as a "raw" interface to system APIs, and
+``platform`` as a higher-level API providing more generally useful values.
+
+
+CI resources
+------------
+
+`Anaconda <https://anaconda.com>`__ has offered to provide physical hardware to
+run Android buildbots.
+
+GitHub Actions is able to host Android emulators on their Linux and macOS
+runners. The free tier currently only provides x86_64 machines; however ARM64
+macOS runners `recently became available on paid plans <https://github.blog/
+2023-10-02-introducing-the-new-apple-silicon-powered-m1-macos-larger-runner-for-github-actions/>`__.
```

Comment:
To clarify - what CI configurations are you proposing would be run? 
* x86_64 runners would be the most readily available; but that would avoid any architecture-specific testing. 
* AWS provides ARM64 Linux instances; can they be used for Android CI testing, or would we need to do ARM64 testing on macOS machines?
* Are there any technical concerns about running CI on emulated machines, rather than physical phone hardware? The iOS situation has the precedent that Apple's own Xcode Cloud offering doesn't provide an on-hardware testing option; they consider simulation to be sufficient. Do we have any analogous basis to limit testing to the Android emulator?

Reply:
I'm not proposing anything specific at the moment, I'm just listing the resources that are available. 

> AWS provides ARM64 Linux instances; can they be used for Android CI testing, or would we need to do ARM64 testing on macOS machines?

macOS would be better, because that's the only OS on which the ARM64 emulator is officially supported. I'll add this to the document.

> Are there any technical concerns about running CI on emulated machines, rather than physical phone hardware?

Since emulators and physical devices use the same ABI, and are running identical or very similar operating system binaries, using an emulator is almost always sufficient. I'll add this to the document.